packages/babel-parser/src/parser/expression.js

```diff
@@ -719,8 +720,19 @@ export default class ExpressionParser extends LValParser {
         state,
         optional,
       );
-    } else if (optional || this.match(tt.bracketL) || this.eat(tt.dot)) {
-      return this.parseMember(base, startPos, startLoc, state, optional);
+    } else if (
+      (computed = this.eat(tt.bracketL)) ||
```

Comment:
Nit: this is kinda ugly. Mind breaking this `else if` into a regular `else` and perform the mutation there?